src/Stratis.Bitcoin.Features.Wallet/WalletRPCController.cs

```diff
@@ -25,20 +29,81 @@ public class WalletRPCController : FeatureController
         /// <summary>Wallet manager.</summary>
         private readonly IWalletManager walletManager;
 
-        public WalletRPCController(IWalletManager walletManager, IFullNode fullNode, IBroadcasterManager broadcasterManager, ILoggerFactory loggerFactory) : base(fullNode: fullNode)
+        /// <summary>Wallet transaction handler.</summary>
+        private readonly IWalletTransactionHandler walletTransactionHandler;
+
+        public WalletRPCController(IWalletManager walletManager, IWalletTransactionHandler walletTransactionHandler, IFullNode fullNode, IBroadcasterManager broadcasterManager, ILoggerFactory loggerFactory) : base(fullNode: fullNode)
         {
             this.walletManager = walletManager;
+            this.walletTransactionHandler = walletTransactionHandler;
             this.fullNode = fullNode;
             this.broadcasterManager = broadcasterManager;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
         }
 
-        [ActionName("sendtoaddress")]
-        [ActionDescription("Sends money to a bitcoin address.")]
-        public uint256 SendToAddress(BitcoinAddress bitcoinAddress, Money amount)
+        [ActionName("walletpassphrase")]
+        [ActionDescription("Stores the wallet decryption key in memory for the indicated number of seconds. Issuing the walletpassphrase command while the wallet is already unlocked will set a new unlock time that overrides the old one.")]
+        [NoTrace]
+        public bool UnlockWallet(string passphrase, int timeout)
+        {
+            Guard.NotEmpty(passphrase, nameof(passphrase));
+
+            // As per RPC method definition this should be the max allowable expiry duration.
+            const int maxDurationInSeconds = 1073741824;
```

Comment:
const at top of class

Reply:
max duration should probably go in the trx handler not in the controller

Reply:
This max duration comes from bitcoin core. I'm not sure how this magic number was derived. Do we even want to restrict this?